src/renderer/gl_engine/tvgGlRenderer.cpp

```diff
@@ -699,6 +699,33 @@ void GlRenderer::prepareCmpTask(GlRenderTask* task, const RenderRegion& vp, uint
 void GlRenderer::endRenderPass(RenderCompositor* cmp)
 {
     auto glCmp = static_cast<GlCompositor*>(cmp);
+    
+    // setup masking and blending render pass configurations
+    if ((glCmp->flags & (tvg::Blending | tvg::Masking)) == (tvg::Blending | tvg::Masking)) {
+        // rearrange render tree
+        auto selfPass = mRenderPassStack.last();
+        mRenderPassStack.pop();
+        auto prevPass = mRenderPassStack.last();
+        mRenderPassStack.pop();
+        auto maskPass = mRenderPassStack.last();
+        mRenderPassStack.pop();
+        mRenderPassStack.push(prevPass);
+        mRenderPassStack.push(maskPass);
+        mRenderPassStack.push(selfPass);
+        // setup composition properties
+        auto prevCompose = mComposeStack.last();
+        auto opacity = glCmp->opacity;
+        auto blendMethod = glCmp->blendMethod;
```

Comment:
```suggestion
        const auto prevCompose = mComposeStack.last();
        const auto opacity = glCmp->opacity;
        const auto blendMethod = glCmp->blendMethod;
```
Since we don’t modify the stored opacity/blend values afterward, making them const could improve clarity and also allow the compiler to optimize better.

Reply:
@hermet what do you think? is it suitable? we did not use approach like this in our code